File: GeneratedEntityReaderClasses/Components/MaterialInventoryComponent.cs
namespace GeneratedNoitaClasses;
public class MaterialInventoryComponent : NoitaComponentBase,  iNoitaType<MaterialInventoryComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool drop_as_item { get; set; } = new();
	public NoitaBool on_death_spill { get; set; } = new();
	public NoitaBool leak_gently { get; set; } = new();
	public NoitaFloat leak_on_damage_percent { get; set; } = new();
	public NoitaFloat leak_pressure_min { get; set; } = new();
	public NoitaFloat leak_pressure_max { get; set; } = new();
	public NoitaFloat min_damage_to_leak { get; set; } = new();
	public NoitaFloat b2_force_on_leak { get; set; } = new();
	public NoitaFloat death_throw_particle_velocity_coeff { get; set; } = new();
	public NoitaBool kill_when_empty { get; set; } = new();
	public NoitaBool halftime_materials { get; set; } = new();
	public NoitaInt do_reactions { get; set; } = new();
	public NoitaBool do_reactions_explosions { get; set; } = new();
	public NoitaBool do_reactions_entities { get; set; } = new();
	public NoitaInt reaction_speed { get; set; } = new();
	public NoitaBool reactions_shaking_speeds_up { get; set; } = new();
	public NoitaDouble max_capacity { get; set; } = new();
	public NoitaVector<NoitaDouble> count_per_material_type { get; set; } = new();
	public NoitaFloat audio_collision_size_modifier_amount { get; set; } = new();
	public NoitaInt last_frame_drank { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		drop_as_item.Read(s);
		on_death_spill.Read(s);
		leak_gently.Read(s);
		leak_on_damage_percent.Read(s);
		leak_pressure_min.Read(s);
		leak_pressure_max.Read(s);
		min_damage_to_leak.Read(s);
		b2_force_on_leak.Read(s);
		death_throw_particle_velocity_coeff.Read(s);
		kill_when_empty.Read(s);
		halftime_materials.Read(s);
		do_reactions.Read(s);
		do_reactions_explosions.Read(s);
		do_reactions_entities.Read(s);
		reaction_speed.Read(s);
		reactions_shaking_speeds_up.Read(s);
		max_capacity.Read(s);
		count_per_material_type.Read(s);
		audio_collision_size_modifier_amount.Read(s);
		last_frame_drank.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		drop_as_item.Write(s);
		on_death_spill.Write(s);
		leak_gently.Write(s);
		leak_on_damage_percent.Write(s);
		leak_pressure_min.Write(s);
		leak_pressure_max.Write(s);
		min_damage_to_leak.Write(s);
		b2_force_on_leak.Write(s);
		death_throw_particle_velocity_coeff.Write(s);
		kill_when_empty.Write(s);
		halftime_materials.Write(s);
		do_reactions.Write(s);
		do_reactions_explosions.Write(s);
		do_reactions_entities.Write(s);
		reaction_speed.Write(s);
		reactions_shaking_speeds_up.Write(s);
		max_capacity.Write(s);
		count_per_material_type.Write(s);
		audio_collision_size_modifier_amount.Write(s);
		last_frame_drank.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/MaterialInventoryComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaDouble
from BasicNoitaTypes import NoitaVector
from BasicNoitaTypes import NoitaDouble

from NoitaStream import NoitaStream
@dataclass
class MaterialInventoryComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    drop_as_item:NoitaBool = field(default_factory=NoitaBool)
    on_death_spill:NoitaBool = field(default_factory=NoitaBool)
    leak_gently:NoitaBool = field(default_factory=NoitaBool)
    leak_on_damage_percent:NoitaFloat = field(default_factory=NoitaFloat)
    leak_pressure_min:NoitaFloat = field(default_factory=NoitaFloat)
    leak_pressure_max:NoitaFloat = field(default_factory=NoitaFloat)
    min_damage_to_leak:NoitaFloat = field(default_factory=NoitaFloat)
    b2_force_on_leak:NoitaFloat = field(default_factory=NoitaFloat)
    death_throw_particle_velocity_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    kill_when_empty:NoitaBool = field(default_factory=NoitaBool)
    halftime_materials:NoitaBool = field(default_factory=NoitaBool)
    do_reactions:NoitaInt = field(default_factory=NoitaInt)
    do_reactions_explosions:NoitaBool = field(default_factory=NoitaBool)
    do_reactions_entities:NoitaBool = field(default_factory=NoitaBool)
    reaction_speed:NoitaInt = field(default_factory=NoitaInt)
    reactions_shaking_speeds_up:NoitaBool = field(default_factory=NoitaBool)
    max_capacity:NoitaDouble = field(default_factory=NoitaDouble)
    count_per_material_type:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaDouble))
    audio_collision_size_modifier_amount:NoitaFloat = field(default_factory=NoitaFloat)
    last_frame_drank:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.drop_as_item.Read(s)
        self.on_death_spill.Read(s)
        self.leak_gently.Read(s)
        self.leak_on_damage_percent.Read(s)
        self.leak_pressure_min.Read(s)
        self.leak_pressure_max.Read(s)
        self.min_damage_to_leak.Read(s)
        self.b2_force_on_leak.Read(s)
        self.death_throw_particle_velocity_coeff.Read(s)
        self.kill_when_empty.Read(s)
        self.halftime_materials.Read(s)
        self.do_reactions.Read(s)
        self.do_reactions_explosions.Read(s)
        self.do_reactions_entities.Read(s)
        self.reaction_speed.Read(s)
        self.reactions_shaking_speeds_up.Read(s)
        self.max_capacity.Read(s)
        self.count_per_material_type.Read(s)
        self.audio_collision_size_modifier_amount.Read(s)
        self.last_frame_drank.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.drop_as_item.Write(s)
        self.on_death_spill.Write(s)
        self.leak_gently.Write(s)
        self.leak_on_damage_percent.Write(s)
        self.leak_pressure_min.Write(s)
        self.leak_pressure_max.Write(s)
        self.min_damage_to_leak.Write(s)
        self.b2_force_on_leak.Write(s)
        self.death_throw_particle_velocity_coeff.Write(s)
        self.kill_when_empty.Write(s)
        self.halftime_materials.Write(s)
        self.do_reactions.Write(s)
        self.do_reactions_explosions.Write(s)
        self.do_reactions_entities.Write(s)
        self.reaction_speed.Write(s)
        self.reactions_shaking_speeds_up.Write(s)
        self.max_capacity.Write(s)
        self.count_per_material_type.Write(s)
        self.audio_collision_size_modifier_amount.Write(s)
        self.last_frame_drank.Write(s)

